index.js

```diff
@@ -27,6 +27,8 @@ var copy = require('copy-to');
 
 module.exports = function (opts) {
   opts = opts || {};
+  var detectJSON = opts.detectJSON;
+  opts.detectJSON = undefined;
```

Comment:
https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#521-objects-that-are-in-hash-table-mode-aka-normalized-objects-dictionary-mode---objects-who-have-a-hash-table-as-a-backing-data-structure-are-not-simple-enumerables